### PR TITLE
Fix env-var naming in docker-compose.yml

### PR DESCRIPTION
**Pull request by mrnicegyu11**
This enables https://github.com/ITISFoundation/osparc-ops-environments/issues/224

## tl;dr

The purpose of this PR is to enable ops in their release-workflows to remove one step/layer, namely the `services/simcore/template.env` file for env-substitution. We will prefer to use the env-vars as given in the `repo.config` files verbatim, without this step in between.

**:warning: If env-vars need to be renamed, this shall always be done in the `environment:` section of the `docker-compose.yml`.**

## details

The way we concatenate and compose the final env-vars of simcore microservices is convoluted due to legacy reasons.
Already today, env-vars can enter the "pipeline" at various stages, including but not limited to the `Dockerfile`, `docker-compose.yml` and variables handed down from the CI Pipeline run config. For a deepdive, see: https://github.com/ITISFoundation/osparc-ops-environments/issues/177 . 

Docker microservices are able to capture the env-vars given from the `repo.config` files in the `configs` branch of https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/tree/configs?ref_type=heads . This should be sufficient, and this PR and associated changes strive to move towards "automagically" capturing env-vars written in the repo.config as the preferred and only way to capture configs.

But, there today exists the file https://github.com/ITISFoundation/osparc-ops-environments/blob/main/services/simcore/template.env as well, which adds the (actually unnecessary) option to rename environment variables (or potentially hardcode them, which luckily is not the case). 

So, in order to purge the `services/simcore/template.env` file, all renaming must be moved to the `docker-compose.yml` file of osparc-simcore. This PR adds the necessary renaming.

### very detailed details

The env-vars that are problematic (`==` contain renaming) in https://github.com/ITISFoundation/osparc-ops-environments/blob/main/services/simcore/template.env  are today:

```
# (1)
DOCKER_IMAGE_TAG=${SIMCORE_IMAGE_TAG}
# (2)
EC2_ACCESS_KEY_ID=${AUTOSCALING_EC2_ACCESS_KEY_ID}
EC2_SECRET_ACCESS_KEY=${AUTOSCALING_EC2_SECRET_ACCESS_KEY}
EC2_REGION_NAME=${AUTOSCALING_EC2_REGION_NAME}
EC2_INSTANCES_ALLOWED_TYPES=${AUTOSCALING_EC2_INSTANCES_ALLOWED_TYPES}
EC2_INSTANCES_AMI_ID=${AUTOSCALING_EC2_INSTANCES_AMI_ID}
EC2_INSTANCES_MACHINES_BUFFER=${AUTOSCALING_EC2_INSTANCES_MACHINES_BUFFER}
EC2_INSTANCES_MAX_INSTANCES=${AUTOSCALING_EC2_INSTANCES_MAX_INSTANCES}
EC2_INSTANCES_SECURITY_GROUP_IDS=${AUTOSCALING_EC2_INSTANCES_SECURITY_GROUP_IDS}
EC2_INSTANCES_SUBNET_ID=${AUTOSCALING_EC2_INSTANCES_SUBNET_ID}
EC2_INSTANCES_KEY_NAME=${AUTOSCALING_EC2_INSTANCES_KEY_NAME}
NODES_MONITORING_NODE_LABELS=${AUTOSCALING_NODES_MONITORING_NODE_LABELS}
NODES_MONITORING_SERVICE_LABELS=${AUTOSCALING_NODES_MONITORING_SERVICE_LABELS}
NODES_MONITORING_NEW_NODES_LABELS=${AUTOSCALING_NODES_MONITORING_NEW_NODES_LABELS}
# (3)
PUBLISHED_HOST_NAME=${MACHINE_FQDN}
# (4)
REGISTRY_URL=${REGISTRY_DOMAIN}
```

re 1: This one requires many changes, including in https://github.com/ITISFoundation/osparc-ops-environments and in *all* deployment-branches of https://git.speag.com/oSparc/osparc-ops-deployment-configuration and the `configs` branch of https://git.speag.com/oSparc/osparc-ops-deployment-configuration . Everywhere, `SIMCORE_IMAGE_TAG` is to be renamed to `DOCKER_IMAGE_TAG`. It requires no changes in `osparc-simcore`. I likely will not open 10 PRs in the ops-repos, so I will commit the needed changes upstream once this PR is merged, if this is ok.

re 2: ~~Here we add the prefix `AUTOSCALING_` in this PR~~ no changes needed, we figured out after discussions with @sanderegg 

re 3: The specific capture `PUBLISHED_HOST_NAME=${MACHINE_FQDN}` is added in this PR on director-v0

re 4: An investigation showed that `REGISTRY_DOMAIN` and `REGISTRY_URL` are aliases and everywhere their value is identical. The alias is set in the `repo.config` in https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/tree/configs?ref_type=heads everywhere. So it is save to be removed without changes.

### CODE REVIEW

**Comment by YuryHrytsuk** [comment]
Make sure it really works (by releasing to NIH Staging) after merging

**Review by GitHK — APPROVED** [review]

**Review by sanderegg — CHANGES_REQUESTED** [review]
ok sorry but I disagree with that one. it is asymetric and confusing. I will propose an alternative and we can discuss.

File: services/docker-compose.yml [hunk 1]
```diff
@@ -72,18 +72,18 @@ services:
...
-      - EC2_INSTANCES_SECURITY_GROUP_IDS=${EC2_INSTANCES_SECURITY_GROUP_IDS}
-      - EC2_INSTANCES_SUBNET_ID=${EC2_INSTANCES_SUBNET_ID}
-      - EC2_INSTANCES_KEY_NAME=${EC2_INSTANCES_KEY_NAME}
+      - EC2_INSTANCES_ALLOWED_TYPES=${AUTOSCALING_EC2_INSTANCES_ALLOWED_TYPES}
+      - EC2_INSTANCES_MACHINES_BUFFER=${AUTOSCALING_EC2_INSTANCES_MACHINES_BUFFER}
+      - EC2_INSTANCES_MAX_INSTANCES=${AUTOSCALING_EC2_INSTANCES_MAX_INSTANCES}
+      - EC2_INSTANCES_NAME_PREFIX=${AUTOSCALING_EC2_INSTANCES_NAME_PREFIX}
+      - EC2_INSTANCES_SECURITY_GROUP_IDS=${AUTOSCALING_EC2_INSTANCES_SECURITY_GROUP_IDS}
+      - EC2_INSTANCES_SUBNET_ID=${AUTOSCALING_EC2_INSTANCES_SUBNET_ID}
+      - EC2_INSTANCES_KEY_NAME=${AUTOSCALING_EC2_INSTANCES_KEY_NAME}
```
[the remark this answers is not in the document]
Comment (mrnicegyu11) [reply services/docker-compose.yml:82]: Very brief:
It is ok if you take over.
I dont know the details of the pydanticsettings-library
I dont today know how `AUTOSCALING_EC2_ACCESS_KEY_ID` works.
If we can remove the `template.env` with your changes I will be happy in either case.
I will stop working on this for now :-)

[the remark this answers is not in the document]
Comment (sanderegg) [reply services/docker-compose.yml:82]: actually after some thinking over it. I think that is ok in the end. will approve it. it is slightly assymetric but I guess that is how that is. since the .env file is gone that should work.

**Review by sanderegg — APPROVED** [review]

**Review by sanderegg — APPROVED** [review]
After some thoughts I think that is actually ok, and should work in the .env-devel since these values are not set there.

**Review by pcrespov — APPROVED** [review]

**Review by bisgaard-itis — APPROVED** [review]
👍🏻
